atomate/vasp/firetasks/parse_outputs.py

```diff
@@ -775,6 +776,590 @@ def run_task(self, fw_spec):
         # a builder to put it into materials collection... -computron
         logger.info("Thermal expansion coefficient calculation complete.")
 
+@explicit_serialize
+class HubbardHundLinRespToDb(FiretaskBase):
+    """
+    Analyze the linear response data generated from get_wf_hubbard_hund_linresp to
+    compute Hubbard U (and Hund J) value(s).
+
+    Required parameters:
+        num_perturb (int): number of perturbed sites
+        spin_polarized (bool):
+        relax_nonmagnetic (bool):
+        db_file (str): path to the db file that holds your tasks
+            collection and that you want to hold the hubbard_hund_linresp
+            collection
+        wf_uuid (str): auto-generated from get_wf_hubbard_hund_linresp,
+            used to make it easier to retrieve task docs
+    """
+
+    required_params = ["num_perturb", "spin_polarized", "relax_nonmagnetic",
+                       "db_file", "wf_uuid"]
+    optional_params = []
+
+    summaries = []
+
+    def run_task(self, fw_spec):
+
+        uuid = self["wf_uuid"]
+        db_file = env_chk(self.get("db_file"), fw_spec)
+        to_db = self.get("to_db", True)
+
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+
+        num_perturb_sites = int(self["num_perturb"])
+        spin_polarized = bool(self["spin_polarized"])
+        relax_nonmagnetic = bool(self["relax_nonmagnetic"])
+
+        keys = ['Ground state', 'NSCF', 'SCF']
+        response_dict = {'Ground state': {}, 'NSCF': {}, 'SCF': {}}
+        perturb_dict = {}
+
+        magnet_order_gs = None
+
+        for key in keys:
+
+            for i in range(num_perturb_sites):
+                response_dict[key].update({'Vup_site'+str(i): [],
```

Comment:
There might be a better way of structuring this. In general, encoding multiple bits of information in a dict key is not ideal, since it then has to be re-parsed later to get information out (eg the perturbed site index here).

An alternative would be a more nested dictionary using the site index as a key too.

Since such a dictionary can be verbose to construct, and since `pydash` is in the `atomate` requirements, an alternative syntax can be:

```
from pydash import set_
response_dict = {}
```

and then, later, when you want to use this you can use:

`_set(response_dict, [key, 'Vdn_site', i], val_to_set)`  # returns {key: 'Vdn_site': i: {val_to_set}

Reply:
That's a great point. I think that I was trying to keep them on the same level as `'magnetic order'`. I'll change that.